index.js

```diff
@@ -60,6 +60,11 @@ function register (server, options, next) {
     reply.continue()
   })
 
+  server.ext('onPreHandler', (request, reply) => {
+    request.logger = (request.logger || logger).child({ req: request })
```

Comment:
Add a `payload` option (or what you prefer) as you were suggesting, and maybe do:

```
request.logger.debug({ payload: req.payload }, 'received payload')
```

This is not doable using a serializers unfortunately.